files/build_templates/docker_image_ctl.j2

```diff
@@ -101,6 +101,19 @@ function preStartAction()
     updateSyslogConf
 }
 
+{%- if docker_container_name == "database" %}
+
+function setPlatformLagIdBoundaries()
+{
+    CHASSIS_CONF=/usr/share/sonic/device/$PLATFORM/chassisdb.conf
+    if [ -f "$CHASSIS_CONF" ]; then
+        source $CHASSIS_CONF
+        $SONIC_DB_CLI CHASSIS_APP_DB SET "SYSTEM_LAG_ID_START" "$lag_id_start"
```

Comment:
@vganesan-nokia We tested this change and it's not working at our side.  $SONIC_DB_CLI CHASSIS_APP_DB SET "SYSTEM_LAG_ID_START" "$lag_id_start"  failed with this error: Invalid database name input : 'CHASSIS_APP_DB'. However,  it worked fine if we do this inside chassisDb docker. E.g.,

docker exec -i ${DOCKERNAME} $SONIC_DB_CLI CHASSIS_APP_DB SET "SYSTEM_LAG_ID_START" "$lag_id_start"

So it seems sonic-db-cli is not aware of CHASSIS_APP_DB when databash.sh is executed.  Any thought on this?

Reply:
@ysmanman, This is supposed to be run only in database-chassis docker in supervisor card only. Assuming you are running this in database-chassis docker in supervisor, please check if you have the CHASSIS_APP_DB defined in database_config.json for redis-chassis instance (i.e. in file /var/run/redis-chassis/sonic-db/database_config.json)

Reply:
@vganesan-nokia I am not suer that this is actually run in database-chassis docker based on the following code:

https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/docker_image_ctl.j2#L174-L178

Line 174-175 are running outside the docker, right?

Reply:
@ysmanman: you are right. It is run outside docker. We can modify the function setPlatfromLagIdBoundaries() to do "docker exec ...". However we do not have to do this or even to run this inside docker. The error you are seeing is because of this CHASSIS_APP_DB not defined in /var/run/redis/sonic-db/database_config.json. We are supposed to have the same content of database_config.json in all instances of /var/run/redis\<instance\>/sonic-db. If CHASSIS_APP_DB is defined in database_config.json with correct instance and if the instance info is specified, sonic-cfggen will connect to correct redis server.

Reply:
@vganesan-nokia We do have CHASSIS_APP_DB defined in database_config.json once switch boots up. However, it appears that /var/run/redis/sonic-db/database_config.json didn't exist yet when setPlatformLagIdBoundaries was invoked.  The file was installed after chassisDb was up. Do you know if SONiC ensures  /var/run/redis/sonic-db/database_config.json is installed before bringing up chassisDb?

Reply:
@ysmanman, The file /var/run/redis\<instance\>/sonic-db/database_config.json is made available before starting the redis server inside the database docker (ref. docker-database-init.sh).  Even though we call the range setting function setPlatformLagIdBoundaries in the script  (ref. database.sh) that starts the database docker, we wait and call this function only after "ping" to the redis-server is successful. So the /var/run/redis\<instance\>/sonic-db/database_config.json should be available when we do this.

Reply:
@vganesan-nokia  Unfortunately, we were observing different from what you explained. The way we set lag id range successfully is to modify setPlatformLagIdBoundaries to set it directly inside chassis database docker like this:

docker exec -i ${DOCKERNAME} $SONIC_DB_CLI CHASSIS_APP_DB SET "SYSTEM_LAG_ID_START" "$lag_id_start"

When setPlatformLagIdBoundaries wass called in the script  (ref. database.sh) that starts the database docker, it seems only  /var/run/redis-chassis/sonic-db/database_config.json was available but not  /var/run/redis/sonic-db/database_config.json.

Reply:
@vganesan-nokia Thanks for checking and confirming!

Reply:
> 
> 
> @vganesan-nokia Thanks for checking and confirming!

@ysmanman, raised issue https://github.com/Azure/sonic-buildimage/issues/7912 and fixed the issue by PR https://github.com/Azure/sonic-buildimage/pull/7911

Reply:
@vganesan-nokia thanks for the update!